tests/posix/pthread_rwlock/src/posix_rwlock.c

```diff
@@ -80,8 +80,14 @@ static void test_rw_lock(void)
 
 	/* Creating N premptive threads in increasing order of priority */
 	for (i = 0; i < N_THR; i++) {
-		pthread_attr_destroy(&attr[i]);
-		pthread_attr_init(&attr[i]);
+
+		ret = pthread_attr_init(&attr[i]);
+		if (ret != 0) {
+			zassert_false(pthread_attr_destroy(&attr[i]),
+					"Unable to destroy pthread object attrib\n");
+			zassert_false(pthread_attr_init(&attr[i]),
```

Comment:
What is this trying to accomplish here?  If the call to pthread_attr_init() fails, the attributed is destroyed and initialized again.  It's not clear to me why this is being performed this way.

Also, the zassert_false() calls do not make much sense: both pthread_attr_destroy() and pthread_attr_init() return an integer error code, not a boolean value.